### PR TITLE
Adding SFCryptoStream

### PR DESCRIPTION
**Pull request by jhneves**
Quick intro:
"Encrypt / decrypt data as it is written / read. Memory consumption is O(1) regardless of cipher data size. It uses buffers in stack memory to perform encryption / decryption in chunks."

@bhariharan @qingqingliu @lqingq @dbreese @MikeNachbaur

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFCryptChunks.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
+//
+//  SFCryptChunks.h
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFCryptChunks.h]: Could you switch the copyright to the standard ```Mobile SDK``` copyright? You should be able to grab it from one of the other files.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFCryptChunks.m [hunk 2]
```diff
@@ -0,0 +1,100 @@
+//
+//  SFCryptChunks.m
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFCryptChunks.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDecryptStream.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+//
+//  SFDecryptStream.h
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDecryptStream.h]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDecryptStream.m [hunk 4]
```diff
@@ -0,0 +1,180 @@
+//
+//  SFDecryptStream.m
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDecryptStream.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFEncryptStream.h [hunk 5]
```diff
@@ -0,0 +1,29 @@
+//
+//  SFEncryptStream.h
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFEncryptStream.h]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFEncryptStream.m [hunk 6]
```diff
@@ -0,0 +1,129 @@
+//
+//  SFEncryptStream.m
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFEncryptStream.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.h [hunk 7]
```diff
@@ -0,0 +1,60 @@
+//
+//  SFCryptoTestUtils.h
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.h]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.m [hunk 8]
```diff
@@ -0,0 +1,83 @@
+//
+//  SFCryptoTestUtils.m
+//  CryptoStream
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamJSONTests.m [hunk 9]
```diff
@@ -0,0 +1,188 @@
+//
+//  SFEncryptDecryptStreamJSONTests.m
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamJSONTests.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamTests.m [hunk 10]
```diff
@@ -0,0 +1,183 @@
+//
+//  SFEncryptDecryptStreamTests.m
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamTests.m]: Could you switch the copyright to the standard Mobile SDK copyright?

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -21,8 +21,17 @@
 /* End PBXAggregateTarget section */
 
 /* Begin PBXBuildFile section */
-		1098078C1CB8713A002AF771 /* UIColor+SFColors.m in Sources */ = {isa = PBXBuildFile; fileRef = E1DDC0FD1CAA2A8B002F51DD /* UIColor+SFColors.m */; };
 		009D77521CA4A92200D5183A /* SFPushNotificationManagerTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 009D77511CA4A92200D5183A /* SFPushNotificationManagerTests.m */; };
+		010A9AE41CC176DD002AF4D3 /* SFCryptChunks.h in Headers */ = {isa = PBXBuildFile; fileRef = 010A9ADE1CC176DD002AF4D3 /* SFCryptChunks.h */; };
+		010A9AE51CC176DD002AF4D3 /* SFCryptChunks.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9ADF1CC176DD002AF4D3 /* SFCryptChunks.m */; };
+		010A9AE61CC176DD002AF4D3 /* SFDecryptStream.h in Headers */ = {isa = PBXBuildFile; fileRef = 010A9AE01CC176DD002AF4D3 /* SFDecryptStream.h */; };
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: We need to make sure that the new files are added to both build targets in the workspace - static and dynamic.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -21,8 +21,17 @@
...
+		010A9AE71CC176DD002AF4D3 /* SFDecryptStream.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9AE11CC176DD002AF4D3 /* SFDecryptStream.m */; };
+		010A9AE81CC176DD002AF4D3 /* SFEncryptStream.h in Headers */ = {isa = PBXBuildFile; fileRef = 010A9AE21CC176DD002AF4D3 /* SFEncryptStream.h */; };
+		010A9AE91CC176DD002AF4D3 /* SFEncryptStream.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9AE31CC176DD002AF4D3 /* SFEncryptStream.m */; };
+		010A9AFC1CC17717002AF4D3 /* SFCryptoStreamTestUtils.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9AF91CC17717002AF4D3 /* SFCryptoStreamTestUtils.m */; };
+		010A9AFD1CC17717002AF4D3 /* SFEncryptDecryptStreamJSONTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9AFA1CC17717002AF4D3 /* SFEncryptDecryptStreamJSONTests.m */; };
+		010A9AFE1CC17717002AF4D3 /* SFEncryptDecryptStreamTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 010A9AFB1CC17717002AF4D3 /* SFEncryptDecryptStreamTests.m */; };
+		1098078C1CB8713A002AF771 /* UIColor+SFColors.m in Sources */ = {isa = PBXBuildFile; fileRef = E1DDC0FD1CAA2A8B002F51DD /* UIColor+SFColors.m */; };
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Also, the new headers here need to be added to the podspec in the repo - ```SalesforceSDKCore.podspec```.

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.h [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+//  CryptoStream
+//
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.h]: Move this file under ```SalesforceSDKCoreTests```.

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.m [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+//
+//  Created by Joao Neves on 4/4/16.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFCryptoStreamTestUtils.m]: Move this file under ```SalesforceSDKCoreTests```.

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamJSONTests.m [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+//  CryptoStream
+//
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamJSONTests.m]: Move this file under ```SalesforceSDKCoreTests```.

File: libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamTests.m [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+//  CryptoStream
+//
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SFEncryptDecryptStreamTests.m]: Move this file under ```SalesforceSDKCoreTests```.

**Comment by bhariharan** [comment]
@jhneves I've added some comments.

**Comment by jhneves** [comment]
@bhariharan updated! Can you take a look again ;-)

**Comment by bhariharan** [comment]
👍

**Pull request merged**